docs/default.nix

```
@@ -4,19 +4,37 @@
   pkgs,
 }:
 let
-  pkgsDoc =
-    import
-      (pkgs.applyPatches {
-        name = "nixpkgs-nixvim-doc";
-        src = pkgs.path;
-        patches = [ ./either_recursive.patch ];
-      })
-      {
-        inherit (pkgs) system;
-        config.allowUnfree = true;
+
+  lib = pkgs.lib.extend (
```

I think a comment would be helpful here. Maybe something along the liens of:

```suggestion
  # Extend nixpkg's lib, so that we can handle recursive leaf types such as `either`
  lib = pkgs.lib.extend (
```

?

---

Yes, this comment would help.